code/game/atoms_movable.dm

```diff
@@ -535,6 +536,37 @@
 			return
 	A.Bumped(src)
 
+/atom/movable/Exited(atom/movable/AM, atom/newLoc)
+	. = ..()
+	if(AM.area_sensitive_contents)
+		for(var/k in get_nested_locs(src) + src)
+			var/atom/movable/location = k
+			LAZYREMOVE(location.area_sensitive_contents, AM.area_sensitive_contents)
+
+/atom/movable/Entered(atom/movable/AM, atom/oldLoc)
+	. = ..()
+	if(AM.area_sensitive_contents)
+		for(var/k in get_nested_locs(src) + src)
+			var/atom/movable/location = k
+			LAZYADD(location.area_sensitive_contents, AM.area_sensitive_contents)
+
+//See traits.dm. Use this in place of ADD_TRAIT.
+/atom/movable/proc/become_area_sensitive(trait_source = TRAIT_GENERIC)
+	if(!HAS_TRAIT(src, TRAIT_AREA_SENSITIVE))
+		RegisterSignal(src, SIGNAL_REMOVETRAIT(TRAIT_AREA_SENSITIVE), .proc/on_area_sensitive_trait_loss)
+		for(var/k in get_nested_locs(src) + src)
+			var/atom/movable/AM = k
```

Comment:
`as anything` and https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#use-descriptive-and-obvious-names